### PR TITLE
fix: prefer inplace softmax to avoid copy

### PR DESCRIPTION
**Pull request by drbh**
This PR modifies `log_softmax` to operate in place, eliminating the need to copy large tensors. This optimization reduces memory consumption during warmup.

For instance, when using `meta-llama/Meta-Llama-3.1-8B-Instruct` on an L4, this change allows running the model with `--max-batch-prefill-tokens` increased from 7192 to 9874 without exceeding memory limits

### CODE REVIEW

**Comment by Narsil** [comment]
> For instance, when using meta-llama/Meta-Llama-3.1-8B-Instruct on an L4, this change allows running the model with --max-batch-prefill-tokens increased from 7192 to 9874 without exceeding memory limits

With chunking in place, we don't want to increase max-batch-prefill-tokens anymore, but instead choose an optimal value :).

That being said, nice find, probably a lot of opportunities for such updates (let's focus on the large tensors though)

**Review by Narsil — APPROVED** [review]

File: server/text_generation_server/models/flash_causal_lm.py [hunk 1]
```diff
@@ -1922,8 +1922,9 @@ def generate_token(
             batch.adapter_meta.adapter_indices = next_adapter_indices
 
         if prefill and prefill_logprobs:
-            # Get prefill logprobs
-            prefill_logprobs_tensor = torch.log_softmax(out, -1)
+            # Get prefill logprobs with inplace softmax (avoid copying the `out` tensor (max_batch_size * vocab_size))
```
Comment (Narsil) [inline server/text_generation_server/models/flash_causal_lm.py:1925]:
```suggestion
            # Get prefill logprobs with inplace softmax (avoid copying the `out` tensor (max_batch_prefill_tokens * vocab_size))
```

There is not batch size anymore per-se.
